bokehjs/src/lib/document/document.ts

```diff
@@ -242,15 +250,30 @@ export class Document implements Equatable {
     }
   }
 
-  /*protected*/ _invalidate_all_models(): void {
-    logger.debug("invalidating document models")
-    // if freeze count is > 0, we'll recompute on unfreeze
-    if (this._all_models_freeze_count === 0) {
+  protected _recompute_timer: number | null = null
+
+  protected _cancel_recompute_all_models(): void {
+    if (this._recompute_timer != null) {
+      clearTimeout(this._recompute_timer)
+      this._recompute_timer = null
```

Comment:
Could this lead to a raise condition? If `_cancel_recompute_all_models` is called multiple times before `clearTimeout` is finished?

Reply:
AFAIK calling `clearTimeout` multiple times for the same ID is just a no-op

Reply:
There is no async code involved where, so no. JavaScript will not stop execution mid a synchronous function.